### PR TITLE
Add docker-compose file and required resources

### PR DESCRIPTION
**Pull request by mayerm94**
Add docker compose and required resources. Run `docker-compose up` to test
Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR. 
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-70
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [ ] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [ ] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+version: '3'
+
+services:
+  endpoint_querier:
+    container_name: endpoint_querier_1
```
Comment (edeyoung) [inline docker-compose.yml]: why `_1`?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - GF_AUTH_ANONYMOUS_ENABLED=true
+      - GF_SECURITY_ALLOW_EMBEDDING=true
+    ports:
+      - "3000:3000"
+
+volumes:
+  pgdata:
+  prometheusData:
+  grafanaData:
+
```
Comment (edeyoung) [inline docker-compose.yml]: do any of these need to be started before the others can start? for example, it looks like `prometheus_postgresql_adapter` is dependent on `pg_prometheus`. if so, should this dependency be stated in this docker compose file so it knows to wait to start the one container until the other one is up?

File: prometheus.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    # metrics_path defaults to '/metrics'
+    # scheme defaults to 'http'.
+    static_configs:
+    - targets: ['endpoint_querier_1:8443']
+
+remote_write:
+    - url: "http://prometheus_postgres_adapter:9201/write"
+remote_read:
+    - url: "http://prometheus_postgres_adapter:9201/read"
+
```
Comment (edeyoung) [inline prometheus.yml]: I don't think this is the task to do it, but a task for next sprint is to figure out what our directory structure for the sub projects and for this top level should be. It's weird to me to have `prometheus.yml` at the top level.

File: .env [hunk 4]
```diff
@@ -0,0 +1 @@
+POSTGRES_PASSWORD=postgrespassword
```
Comment (edeyoung) [inline .env]: I'm curious about this. let's chat about it.

File: docker-compose.yml (hunk 1 above)
Comment (mayerm94) [reply docker-compose.yml]: I thought this will better prepare us for when we have multiple endpoint queriers

File: .env (hunk 4 above)
Comment (mayerm94) [reply .env]: There are a couple of other patterns we could use for this, see: https://docs.docker.com/compose/environment-variables/

File: docker-compose.yml (hunk 2 above)
Comment (mayerm94) [reply docker-compose.yml]: Based on https://docs.docker.com/compose/startup-order/ I was under the impression that the order of startup is determined by ``links:``. However I just read the docs for `links` and although it does influence startup order it was deprecated in the previous version of docker-compose. I will switch to using `depends_on` to determine startup order. All of the containers join the same default network in the current version of docker, so the behavior of the `link` will be preserved.

File: .env (hunk 4 above)
Comment (edeyoung) [reply .env]: great link, thanks!

I now see that this is a default and that system environment variables will override it. perfect.

File: prometheus.yml (hunk 3 above)
Comment (edeyoung) [reply prometheus.yml]: added https://oncprojectracking.healthit.gov/support/browse/LANTERN-87

File: README.md [hunk 5]
```diff
@@ -21,6 +38,15 @@ To start the Docker container that you just bult run:
 ```bash
 docker run -p 8443:8443 -it endpoint_querier
 ```
+# Additional Services
+The data collected by the endpoint querier can then be collected by Prometheus, which can be written to a Postgres database using the Prometheus Postgres storage adapter. This data can ultimately be viewed in Grafana. Below is information about how to start these additional services.
+
+## Starting All Services Using docker-compose
+All of the required services to run the Lantern back end are contained in the docker-compose file.
+```bash
+docker-compose up
```
Comment (edeyoung) [inline README.md]: I'd add a few other pointers, particularly pointing out the different between `down` and `stop`:

link to docker-compose docs: https://docs.docker.com/compose/reference
`docker-compose up -d` runs everything in the background
`docker-compose stop` stops everything but keeps the containers, volumes, etc.
`docker-compose start` restarts containers after having called `docker-compose stop`
`docker-compose down` stops containers and removes them as well as networks (looks like you can remove volumes but it's not by default).

Comment (edeyoung) [reply README.md]: needed to modify grafana instructions to use `pg_prometheus:5432`

File: .env (hunk 4 above)
Comment (ttaylor249) [inline .env]: As a general rule, you should not be including passwords in files that are committed to version control.  Instead, document that that environment variable needs to be defined, and require it to be defined prior to running compose.

Comment (ttaylor249) [reply .env]: I think what I would recommend is renaming this to env.sample, then add a note in the README.md file that the env.sample file should be renamed to .env, and edited as appropriate.  Make sure that .env is added to the .gitignore file so that it's not committed to version control again.
